include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2454,595 +2470,567 @@ namespace internal
-                    {
-                      const unsigned int i_ = reorientate(0, i);
-
-                      const unsigned int ind1 = index_array_hermite[0][2 * i] *
-                                                VectorizedArrayType::size();
-                      const unsigned int ind2 =
-                        index_array_hermite[0][2 * i + 1] *
-                        VectorizedArrayType::size();
-                      for (unsigned int comp = 0; comp < n_components; ++comp)
-                        function_2a(
-                          temp1[i_ + 2 * comp * dofs_per_face],
-                          temp1[i_ + dofs_per_face + 2 * comp * dofs_per_face],
-                          global_vector_ptr + ind1 +
-                            comp * static_dofs_per_component *
-                              VectorizedArrayType::size() +
-                            dof_info.component_dof_indices_offset
-                                [active_fe_index][first_selected_component] *
-                              VectorizedArrayType::size(),
-                          global_vector_ptr + ind2 +
-                            comp * static_dofs_per_component *
-                              VectorizedArrayType::size() +
-                            dof_info.component_dof_indices_offset
-                                [active_fe_index][first_selected_component] *
-                              VectorizedArrayType::size(),
-                          grad_weight,
-                          indices,
-                          indices);
-                    }
-                  else
+                  for (unsigned int i = 0; i < dofs_per_face; ++i)
                     {
-                      Assert(false, ExcNotImplemented());
-
-                      const unsigned int n_filled_lanes =
-                        dof_info
-                          .n_vectorization_lanes_filled[dof_access_index][cell];
+                      if (n_face_orientations == 1)
+                        {
+                          const unsigned int i_ = reorientate(0, i);
 
-                      for (unsigned int v = 0; v < n_filled_lanes; ++v)
-                        for (unsigned int comp = 0; comp < n_components; ++comp)
-                          {
-                            const unsigned int i_ = reorientate(0, i);
-                            function_3a(temp1[i_ + 2 * comp * dofs_per_face][v],
+                          const unsigned int ind1 =
+                            index_array_hermite[0][2 * i] *
+                            VectorizedArrayType::size();
+                          const unsigned int ind2 =
+                            index_array_hermite[0][2 * i + 1] *
+                            VectorizedArrayType::size();
+                          for (unsigned int comp = 0; comp < n_components;
+                               ++comp)
+                            function_2a(temp1[i_ + 2 * comp * dofs_per_face],
```

Comment:
I assume that you checked that you need to reorient the local array `temp1` rather than in the access of the `global_vector`? I never remember which way around the indices are.

Reply:
I have done it analogy to:

https://github.com/dealii/dealii/blob/379293c42a11e28a89dd733c77c53593e3b93179/include/deal.II/matrix_free/evaluation_kernels.h#L3089-L3094